src/db_state.rs

```diff
@@ -139,6 +140,7 @@ pub(crate) struct DbState {
     memtable: WritableKVTable,
     wal: WritableKVTable,
     state: Arc<COWDbState>,
+    error: WriteOnceRegister<SlateDBError>,
```

Comment:
does using `OnceCell<SlateDBError>` appropriate on this use case?

Reply:
I had a looked into it. The thing that's missing is the ability to await a value being written to the cell (`OnceLock` supports this but for threads). I suppose we could use https://docs.rs/async-once-cell/latest/async_once_cell/. It didn't seem worth adding a dependency.

We use the ability to await a value in a few places:
- The writer clients await a `Result((), SlateDbError)` being set on the `durable` register of a wal/l0 memtable being set before acknowledging a write when `await_durable: true`.
- The writer clients await a `Result((), SlateDbError)` being set on the `flushed` register of a l0 memtable being set when applying backpressure.
- The memtable flush and wal flush tasks wait on the global error register being set as part of their core `select!`. If an error is set, then they exit (and set the result in all wal/l0 memtables)

Reply:
i see, so this error is not only a marker, but also a channel used in `select!`. thank you for the explanation.

Reply:
Yea, I went down this rabbithole when working on the batch write stuff. The best I could find was the sync watch that @rodesai is using.